app/controllers/catalog_controller.rb

```diff
@@ -317,14 +318,15 @@ def collection_search
     search[:total] = @response.response['numFound']
     session[:search] = search
   end
-
+  
   def metadata_only_collections
     meta_colls = []
-    other_rights_fquery = '(otherRights_tesim:localDisplay OR otherRights_tesim:metadataDisplay)'
-    params = { fq: "{!join from=collections_tesim to=id}#{other_rights_fquery}", q: 'type_tesim:Collection', rows: 300 }
+    params = { fq: "{!join from=collections_tesim to=id}#{metadata_only_fquery}", q: 'type_tesim:Collection', rows: 300 }
     response = raw_solr(params)
     response.docs.each do |doc|
-      meta_colls << doc['id_t']
+      mix_obj = mix_objects?(doc['id_t'])
```

Comment:
Will the recurring checks with Solr queries for all metadata_only collection slowdown the page loading or not? Is it possible to do it in the page level?

Reply:
@lsitu - I think it will slow down the page loading but not much because we don't have that much metadata_only_collection.  Do you have any suggestion of how to do it in the page level?  Thanks

Reply:
What causing my attention is that if we just have 10 such collections, we need to run 20 SOLR queries for each page loading, even with pages that are all objects. That could be a huge burden for our servers.
Maybe we can just check that with the records in search results in each page to see whether it contains metadata-only collections, and then just check for those metadata-only collections instead?
https://github.com/ucsdlib/damspas/pull/494/files#diff-956e46b33fb5307990c4e1a4b5fd86ccR311
https://github.com/ucsdlib/damspas/pull/494/files#diff-956e46b33fb5307990c4e1a4b5fd86ccR274